docs/proposal/2022-12-23-rbac.md

```diff
@@ -0,0 +1,399 @@
+            mapping:
+              type: UserEmail # user email will be used as identifying subject without group
+              usernamePrefix: "" # prefix added to the user email fetched from comm platform
+              groupsPrefix: "" # not used
+
+            # Option 2: user impersonation using email address and channel name as a group
+            mapping:
+              type: ChannelName # channel name will be used as identifying subject
+              usernamePrefix: "" # prefix added to the user email fetched from comm platform
+              groupsPrefix: "" # added to the channel name
+
+            # Option 3: user impersonation using email address and communication platform user group as a group
+            mapping:
+              type: UserGroupName # user group name will be used as identifying subject
+              usernamePrefix: "" # prefix added to the user email fetched from comm platform
+              groupsPrefix: "" # added to the user group name
+
+            # Option 4 (default): static impersonation
+            mapping:
+              type: Static # impersonate as a given group or user every time a given executor is executed
+              username: "any" # username to impersonate
+              groups: [ "developers" ] # groups to impersonate
+
+      config:
+        defaultNamespace: default
+        restrictAccess: false
+
+  'kubectl-pods-rw':
+    botkube/kubectl:
+      enabled: true
+      context:
+        kubeconfig: &kubeconfigCtx # no need to specify it again if it's the same as above
+  'kubectl-deploy-rw':
+    botkube/kubectl@v1.0.0:
+      enabled: true
+      context:
+        kubeconfig: &kubeconfigCtx # no need to specify it again if it's the same as above
+```
+
+However, allowing mapping user and group separately gives us more flexibility. For example, we will be able to implement username mapping for users (for those who don't want to share email addresses with Botkube), and still getting group name from the channel name.
+
+### Executing commands with proper access rights
+
+After identifying the user and/or group, instead of introspection, we could run executor as a short-living separate Pod with a dedicated ServiceAccount and (Cluster)RoleBinding. This way we could use Kubernetes RBAC to restrict access to the cluster, with the isolation provided by Kubernetes. However, this is not possible as our plugin system is not compatible:
+
+> While the plugin system is over RPC, it is currently only designed to work over a local reliable network. Plugins over a real network are not supported and will lead to unexpected behavior.
+
+**Source:** https://github.com/hashicorp/go-plugin
+
+## Consequences
+
+Once the proposal is accepted, the following changes will be made:
+
+1. Update configuration:
+
+   - Introduce `executors[name][pluginname]context` field.
+   - Support initially:
+
+     - static mapping for both users and groups,
+     - email-based user mapping,
```

Comment:
I would also push that into the follow-up task. It would be good to know if our clients want to give us a `users:read` and `users:read.email` for their workspace.

Additionally, this will be platform-specific and would require more work.

Reply:
So for users you would suggest just static mapping? That sounds good to me, as long as Blair agrees with that 👍